src/app/rengular-components-schema.ts

```diff
@@ -83,6 +83,11 @@ const RENGULAR_REGISTRY: {
   'http://rengular.js.org/schema/ChoiceMenu': {
     component: ChoiceMenuComponent,
     inputs: {
+      'http://rengular.js.org/schema/ChoiceMenu#options': (component, data) => {
+        component.choices = data.map(tuple => ({
+          gridCols: tuple['http://rengular.js.org/schema/gridCols']['@id'],
+        } as ActionOptDef));
+      },
```

Comment:
Follow your definition, the data json would become:

```json
{
  "@type": "http://rengular.js.org/schema/ChoiceMenu",
  "http://rengular.js.org/schema/ChoiceMenu#choices": [...buttons...],
  "http://rengular.js.org/schema/ChoiceMenu#options": [
    {
      "@type": "....The type of the gridCols ...",
      "http://rengular.js.org/schema/gridCols": {
        "@id": 2
      }
    }
  ]
}
```

Then why not make it concise without decreasing the intuition:

```json
{
  "@type": "http://rengular.js.org/schema/ChoiceMenu",
  "http://rengular.js.org/schema/ChoiceMenu#choices": [...buttons...],
  "http://rengular.js.org/schema/gridCols": 2,
}
```